.golangci.yml

```
@@ -0,0 +1,103 @@
+run:
+  timeout: 2m
+
+  # Skip the eth bindings.
+  skip-dirs:
+    - backend/ethereum/bindings/
+
+  # Do not change go.{mod,sum}.
+  modules-download-mode: readonly
+
+linters:
+  enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - dupl
+    - errcheck
+    - exportloopref
+    - funlen
+    # We use init() functions.
+    #- gochecknoinits
+    - goconst
+    - gocritic
+    - gocyclo
+    - gofmt
+    - goimports
+    # This is stupid since it reports every constant as magic-number.
+    #- gomnd
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+    # This thing reports long lines, but we have a lot of them…
+    #- lll
+    - misspell
+    # We often just have a 'return' when using named returns.
+    #- nakedret
+    - nolintlint
+    - rowserrcheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+    - whitespace
+    - asciicheck
+    # We often use globals for backend stuff
+    #- gochecknoglobals
+    - gocognit
+    - godot
+    # Reports TODO/BUG/FIXME
+    #- godox
+    # Keeps reporting 'do not define dynamic errors, use wrapped static errors instead'.
+    #- goerr113
+    # We don't care about alignment right now.
+    #- maligned
+    - nestif
+    - prealloc
+    # This always reports that tests should be Black Box.
+    #- testpackage
+    #- wsl
+
+issues:
+  exclude-rules:
+    - path: pkg/test
+      linters:
+        - stylecheck
+
+    - path: test
```

so what exactly does `path: test` match?

---

It matches the string `test` on the relative path of the file.  
So `sync/mutex_test.go` or `test/eventually.go` match.

---

Then we should update this to something like
```suggestion
    - path: _test.go
```
because we only want to exclude test files from below linter checks.

We still want the full linting suite to run for testing helpers in `test/...`.

Same for the text three exclusion rules, which also use `path: test`.

---

Im currently updating the linter config in https://github.com/hyperledger-labs/go-perun/issues/132. So we either merge the current state of wait til the update is done and then use it here as well.

---

Happy with either, what's easier for you. If you rather want to merge now and fix all remaining open discussions in a follow-up, please collect the open discussions in an issue.

---

Added it to as TODO for https://github.com/hyperledger-labs/go-perun/issues/132.